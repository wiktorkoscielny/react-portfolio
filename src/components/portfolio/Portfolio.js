import React from 'react'
import './portfolio.css'

// images
import IMG1 from '../../assets/image-1.jpg'
import IMG2 from '../../assets/image-2.jpg'
import IMG3 from '../../assets/image-3.jpg'
import IMG4 from '../../assets/image-4.jpg'
import IMG5 from '../../assets/image-5.jpg'
import IMG6 from '../../assets/image-6.jpg'

// generating portfolio items using map method

const data = [
  {
    id: 1,
    image: IMG1,
    dataTestId: 'image-1',
    title: 'This is a portfolio item title',
    alt: 'Portfolio picture no. 1',
    github: 'htpps://github.com',
    demo: 'https://www.youtube.com',
  },
  {
    id: 2,
    image: IMG2,
    dataTestId: 'image-2',
    title: 'This is a portfolio item title',
    alt: 'Portfolio picture no. 2',
    github: 'htpps://github.com',
    demo: 'https://www.youtube.com',
  },
  {
    id: 3,
    image: IMG3,
    dataTestId: 'image-3',
    title: 'This is a portfolio item title',
    alt: 'Portfolio picture no. 3',
    github: 'htpps://github.com',
    demo: 'https://www.youtube.com',
  },
  {
    id: 4,
    image: IMG4,
    dataTestId: 'image-4',
    title: 'This is a portfolio item title',
    alt: 'Portfolio picture no. 4',
    github: 'htpps://github.com',
    demo: 'https://www.youtube.com',
  },
  {
    id: 5,
    image: IMG5,
    dataTestId: 'image-5',
    title: 'This is a portfolio item title',
    alt: 'Portfolio picture no. 5',
    github: 'htpps://github.com',
    demo: 'https://www.youtube.com',
  },
  {
    id: 6,
    image: IMG6,
    dataTestId: 'image-6',
    title: 'This is a portfolio item title',
    alt: 'Portfolio picture no. 6',
    github: 'htpps://github.com',
    demo: 'https://www.youtube.com',
  }
]

const Portfolio = () => {
  return (
    <section id='portfolio'>
      <h5>My recent work</h5>
      <h2>Portfolio</h2>

      <div className='container portfolio__container'>

        {
          data.map(({id, image, dataTestId, title, alt, github, demo}) => {
            return (
              <article key={id} className='portfolio__item'>
                <div className='portfolio__item-image'>
                  <img data-testid={dataTestId} src={image} alt={alt}></img>
                </div>
                <h3>{title}</h3>
                <div className='portfolio__item-cta'>
                  <a href={github} className='btn' target='_blank'>Github</a>
                  <a href={demo} className='btn btn-primary' target='_blank'>Live Demo</a>
                </div>
              </article>
            )
          })
        }
        
      </div>

    </section>
  )
}

export default Portfolio