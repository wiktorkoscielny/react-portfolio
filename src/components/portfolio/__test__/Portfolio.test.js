import React from "react";
import { unmountComponentAtNode } from "react-dom";
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
// component
import Portfolio from './../Portfolio';

let container = null;
beforeEach(()=>{
    container = document.createElement('div');
    document.body.appendChild(container);
});
afterEach(()=>{
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});
it('renders without crashing', ()=>{
    render(<Portfolio />, container);
});
// img
it('Check if imgs are rendering properly', () => {
    const {getByTestId} = render(<Portfolio />);
    expect(getByTestId('image-1')).toBeVisible();
    expect(getByTestId('image-2')).toBeVisible();
    expect(getByTestId('image-3')).toBeVisible();
    expect(getByTestId('image-4')).toBeVisible();
    expect(getByTestId('image-5')).toBeVisible();
    expect(getByTestId('image-6')).toBeVisible();
});
// snapshot
it('matches snapshot', ()=>{
    let tree = renderer.create(<Portfolio />).toJSON();
    expect(tree).toMatchSnapshot();
});


    


