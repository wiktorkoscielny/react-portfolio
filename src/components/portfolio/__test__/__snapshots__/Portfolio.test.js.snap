// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  id="portfolio"
>
  <h5>
    My recent work
  </h5>
  <h2>
    Portfolio
  </h2>
  <div
    className="container portfolio__container"
  >
    <article
      className="portfolio__item"
    >
      <div
        className="portfolio__item-image"
      >
        <img
          alt="Portfolio picture no. 1"
          data-testid="image-1"
          src="image-1.jpg"
        />
      </div>
      <h3>
        This is a portfolio item title
      </h3>
      <div
        className="portfolio__item-cta"
      >
        <a
          className="btn"
          href="htpps://github.com"
          target="_blank"
        >
          Github
        </a>
        <a
          className="btn btn-primary"
          href="https://www.youtube.com"
          target="_blank"
        >
          Live Demo
        </a>
      </div>
    </article>
    <article
      className="portfolio__item"
    >
      <div
        className="portfolio__item-image"
      >
        <img
          alt="Portfolio picture no. 2"
          data-testid="image-2"
          src="image-2.jpg"
        />
      </div>
      <h3>
        This is a portfolio item title
      </h3>
      <div
        className="portfolio__item-cta"
      >
        <a
          className="btn"
          href="htpps://github.com"
          target="_blank"
        >
          Github
        </a>
        <a
          className="btn btn-primary"
          href="https://www.youtube.com"
          target="_blank"
        >
          Live Demo
        </a>
      </div>
    </article>
    <article
      className="portfolio__item"
    >
      <div
        className="portfolio__item-image"
      >
        <img
          alt="Portfolio picture no. 3"
          data-testid="image-3"
          src="image-3.jpg"
        />
      </div>
      <h3>
        This is a portfolio item title
      </h3>
      <div
        className="portfolio__item-cta"
      >
        <a
          className="btn"
          href="htpps://github.com"
          target="_blank"
        >
          Github
        </a>
        <a
          className="btn btn-primary"
          href="https://www.youtube.com"
          target="_blank"
        >
          Live Demo
        </a>
      </div>
    </article>
    <article
      className="portfolio__item"
    >
      <div
        className="portfolio__item-image"
      >
        <img
          alt="Portfolio picture no. 4"
          data-testid="image-4"
          src="image-4.jpg"
        />
      </div>
      <h3>
        This is a portfolio item title
      </h3>
      <div
        className="portfolio__item-cta"
      >
        <a
          className="btn"
          href="htpps://github.com"
          target="_blank"
        >
          Github
        </a>
        <a
          className="btn btn-primary"
          href="https://www.youtube.com"
          target="_blank"
        >
          Live Demo
        </a>
      </div>
    </article>
    <article
      className="portfolio__item"
    >
      <div
        className="portfolio__item-image"
      >
        <img
          alt="Portfolio picture no. 5"
          data-testid="image-5"
          src="image-5.jpg"
        />
      </div>
      <h3>
        This is a portfolio item title
      </h3>
      <div
        className="portfolio__item-cta"
      >
        <a
          className="btn"
          href="htpps://github.com"
          target="_blank"
        >
          Github
        </a>
        <a
          className="btn btn-primary"
          href="https://www.youtube.com"
          target="_blank"
        >
          Live Demo
        </a>
      </div>
    </article>
    <article
      className="portfolio__item"
    >
      <div
        className="portfolio__item-image"
      >
        <img
          alt="Portfolio picture no. 6"
          data-testid="image-6"
          src="image-6.jpg"
        />
      </div>
      <h3>
        This is a portfolio item title
      </h3>
      <div
        className="portfolio__item-cta"
      >
        <a
          className="btn"
          href="htpps://github.com"
          target="_blank"
        >
          Github
        </a>
        <a
          className="btn btn-primary"
          href="https://www.youtube.com"
          target="_blank"
        >
          Live Demo
        </a>
      </div>
    </article>
  </div>
</section>
`;
